corehq/apps/geospatial/static/geospatial/js/case_grouping_map.js

```diff
@@ -327,6 +336,105 @@ hqDefine("geospatial/js/case_grouping_map",[
         );
     }
 
+    function caseGroupSelectModel() {
+        'use strict';
+        var self = {};
+
+        self.allGroups = ko.observableArray([]);
+        self.allCaseGroups;
+        self.visibleGroupIDs = ko.observableArray([]);
```

Comment:
Not sure if it's difficult to refactor now, but a better way would be to have an attr called `is_visible` and only have groups instead of the double allGroups and visibleGroups.

Reply:
Not sure if I'm misunderstanding your suggestion, but `allGroups` is used to keep track of all possible groups, regardless of their visibility. `visibleGroups` keeps track of the visible ones.